service/frontend/workflow_handler.go

```diff
@@ -4476,6 +4480,41 @@ func (wh *WorkflowHandler) validateTaskQueue(t *taskqueuepb.TaskQueue) error {
 	return nil
 }
 
+func (wh *WorkflowHandler) validateVersionRuleBuildId(request *workflowservice.UpdateWorkerVersioningRulesRequest) error {
+	validateBuildId := func(bid string) error {
+		if len(bid) > 255 {
+			return serviceerror.NewInvalidArgument(fmt.Sprintf("BuildId must be <= 255 characters, was %d", len(bid)))
+		}
+
+		if err := common.ValidateUTF8String("BuildId", bid); err != nil {
```

Comment:
I'd follow the suggestion in this case, but also fine to leave it open for other checks.

Note that in the future we'll be able to remove all these checks since proto will do it. But good to add for now.

Reply:
are you suggesting a change here? not sure what you mean by "follow the suggestion, but also fine to leave it open for other checks"

Reply:
The linter suggestion was just to return the last error value directly, right? I just meant I would do that but it's also fine to leave it how it is (i.e. so you can insert other checks without changing any code)

Reply:
yeah that was the linter suggestion. makes sense